.ci/scripts/get_compatible_scipy_version.py

```diff
@@ -14,10 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ===============================================================================
-from sys import exit, stderr
+from sys import argv, exit, stderr
 from sys import version_info as python_version
 
-from daal4py.sklearn._utils import sklearn_check_version
+if len(argv) == 1:
+    from daal4py.sklearn._utils import sklearn_check_version
+else:
+    # This is necessary to prevent import errors from sklearnex and by extension
+    # sklearn caused by initial scipy and/or numpy versions that are installed.
+    # It is written in a way to be `packaging` independent. This branch occurs
+    # when a sklearn version is given to the script externally.
+    def sklearn_check_version(ver):
+        ver = [int(i) if i != "" else 0 for i in ver.split(".")]
+        sk_ver = [int(i) if i != "" else 0 for i in str(argv[1]).split(".")]
+        return sk_ver[0] > ver[0] or (sk_ver[0] == ver[0] and sk_ver[1] >= ver[1])

```

Comment:
Ah yeah, I notice that sklearn had a 0.22.2.post1 which would cause this to fail. I have adapted it to only convert the first two values to ints, as is only necessary for the matching.